#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#          name                   paramtype level desciption                    default=? min  max
gen.add("Kp_pos", double_t, 0, "Kp_pos", 80, 0, 400)#
gen.add("Kp_ori", double_t, 0, "Kp_ori", 8, 0, 20)#
gen.add("Kv_pos", double_t, 0, "Kv_pos", 2, 0, 200)#
gen.add("Kv_ori", double_t, 0, "Kv_ori", 1, 0, 10)#

gen.add("Md", double_t, 0, "inertia", 3, 0, 100)#
gen.add("Bd", double_t, 0, "damping", 8, 0, 100)
gen.add("Kd", double_t, 0, "stiffness", 4, 0, 100)
#pkgname nodename name
exit(gen.generate(PACKAGE, "dynamic_compliance", "nullspace_impedance_MBobserver_controller_param"))
